File: javalin-openapi-processor/src/main/java/net/dzikoysk/openapi/processor/processing/DefaultVisitor.java
package net.dzikoysk.openapi.processor.processing;

import javax.lang.model.element.AnnotationMirror;
import javax.lang.model.element.AnnotationValue;
import javax.lang.model.element.AnnotationValueVisitor;
import javax.lang.model.element.VariableElement;
import javax.lang.model.type.TypeMirror;
import java.util.List;

public abstract class DefaultVisitor<R, P> implements AnnotationValueVisitor<R, P> {

    @Override
    public R visit(AnnotationValue av, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visit(AnnotationValue av) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitBoolean(boolean b, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitByte(byte b, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitChar(char c, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitDouble(double d, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitFloat(float f, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitInt(int i, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitLong(long i, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitShort(short s, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitString(String s, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitType(TypeMirror t, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitEnumConstant(VariableElement c, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitAnnotation(AnnotationMirror a, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitArray(List<? extends AnnotationValue> vals, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

    @Override
    public R visitUnknown(AnnotationValue av, P p) {
        throw new UnsupportedOperationException("Not implemented");
    }

}

File: javalin-openapi-processor/src/main/kotlin/net/dzikoysk/openapi/processor/processing/DefaultVisitor.kt
package net.dzikoysk.openapi.processor.processing

import javax.lang.model.element.AnnotationMirror
import javax.lang.model.element.AnnotationValue
import javax.lang.model.element.AnnotationValueVisitor
import javax.lang.model.element.VariableElement
import javax.lang.model.type.TypeMirror

abstract class DefaultVisitor<R, P> : AnnotationValueVisitor<R, P> {

    override fun visit(av: AnnotationValue, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visit(av: AnnotationValue): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitBoolean(b: Boolean, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitByte(b: Byte, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitChar(c: Char, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitDouble(d: Double, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitFloat(f: Float, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitInt(i: Int, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitLong(i: Long, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitShort(s: Short, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitString(s: String, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitType(t: TypeMirror, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitEnumConstant(c: VariableElement, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitAnnotation(a: AnnotationMirror, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitArray(vals: List<AnnotationValue>, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

    override fun visitUnknown(av: AnnotationValue, p: P): R {
        throw UnsupportedOperationException("Not implemented")
    }

}